shopinvader/models/res_partner.py

```diff
@@ -104,29 +209,28 @@ def write(self, vals):
     def addr_type_display(self):
         return self._fields["address_type"].convert_to_export(self.address_type, self)
 
-    def action_enable_for_shop(self):
-        self.write({"shopinvader_enabled": True})
-        # TODO: maybe better to hook to an event?
-        for partner in self:
-            if partner.address_type == "profile":
-                notif_type = "customer_validated"
-                backends = partner.mapped("shopinvader_bind_ids.backend_id")
-                recipient = partner
-            elif partner.address_type == "address":
-                notif_type = "address_validated"
-                backends = partner.mapped("parent_id.shopinvader_bind_ids.backend_id")
-                recipient = partner.parent_id
-            recipient._shopinvader_notify(backends, notif_type)
-            name = partner.name or partner.contact_address.replace("\n", " | ")
-            msg_body = _("Shop {addr_type} '{name}' validated").format(
-                addr_type=partner.addr_type_display().lower(), name=name
-            )
-            recipient.message_post(body=msg_body)
-
-    def _shopinvader_notify(self, backends, notif_type):
-        self.ensure_one()
-        for backend in backends:
-            backend._send_notification(notif_type, self)
+    def action_shopinvader_validate_customer(self):
+        return self.shopinvader_bind_ids.action_shopinvader_validate()
+
+    def action_shopinvader_validate_address(self):
+        wiz = self._get_shopinvader_validate_address_wizard()
+        action = self.env.ref("shopinvader.shopinvader_address_validate_act_window")
+        action_data = action.read()[0]
+        action_data["res_id"] = wiz.id
+        return action_data
+
+    def _get_shopinvader_validate_address_wizard(self, **kw):
+        ids = self.ids
+        if self.env.context.get("validate_all"):
+            ids = self.child_ids.filtered_domain(
```

Comment:
`validate_all` is kind of misleading here. It's in fact validating all children, but it's ignoring the actual partners
I'd either change the key for `validate_children` or make sure all ids are processed (self + children)

Reply:
I change the name